test/Microsoft.Azure.WebJobs.Host.EndToEndTests/EventHubEndToEndTests.cs

```diff
@@ -127,41 +129,107 @@ public static void SendEvents_TestHub2(int numEvents, string input, [EventHub(Te
                 }
             }
         }
+    }
+
+    public class EventHubPartitionKeyEndToEndTests : IClassFixture<TestFixture<EventHubPartitionKeyEndToEndTests.EventHubTestJobs>>
```

Comment:
why are you creating a separate end to end test suite for this partition test? Did you run into problems? We shouldn't need a separate full test suite for each new test scenario we want to add.

Reply:
yeah, I want to reuse a eventhub for the test and I do not want create a new eventhub. If I add everything in one class another function can be triggered on new event.

Reply:
Well if you consume all the events after each test that would solve the issue, right? Ensuring the checkpoint is moved to the "end". Even with your new test suite, it seems to me that you can get leakage across suites. E.g. if one suite wrote a bunch of events that didn't get fully processed (e.g. due to failures or something, or because the tests passed but didn't consume all events), the other suite will see them in the hub when it runs?

Reply:
Yes, the other suite will see unprocessed messages. So you propose just to create the other eventhub for the test?

Reply:
We don’t want a new event hub per test, nor do we want a new test suite per test :) We need to find a way to clear/consume the messages before each test so each test starts clean.

Reply:
I fixed the tests:
1. Use one test suite.
2. Use one eventhub.
3. Filter messages for the current test is using.